deploy/deploy-centralvenv.sh

```diff
@@ -0,0 +1,1000 @@
+    #       because those are deployment flavor dependent (e.g. prod, preprod, test)
+    [[ -d ${wmCurrPath}/auth/ ]] || mkdir -p ${wmCurrPath}/auth || return $?
+    for service in $enabledList
+    do
+        [[ -d ${wmCurrPath}/auth/${service} ]] || mkdir -p ${wmCurrPath}/auth/${service} || { err=$?; echo "could not create auth path for: $service";  return $err  ;}
+        _addWMCoreVenvVar PYTHONPATH ${wmCurrPath}/auth/${service}:$PYTHONPATH
+
+        [[ -d ${wmStatePath}/${service} ]] || mkdir -p ${wmStatePath}/${service} || { err=$?; echo "could not create state path for: $service";  return $err  ;}
+        [[ -d ${wmLogsPath}/${service} ]] || mkdir -p ${wmLogsPath}/${service} || { err=$?; echo "could not create logs path for: $service";  return $err  ;}
+    done
+
+    _addWMCoreVenvVar X509_USER_CERT ${wmAuthPath}/dmwm-service-cert.pem
+    _addWMCoreVenvVar X509_USER_KEY ${wmAuthPath}/dmwm-service-key.pem
+    _addWMCoreVenvVar WMCORE_SERVICE_CONFIG ${wmCfgPath}
+    _addWMCoreVenvVar WMCORE_SERVICE_ENABLED ${wmEnabledPath}
+    _addWMCoreVenvVar WMCORE_SERVICE_AUTH ${wmAuthPath}
+    _addWMCoreVenvVar WMCORE_SERVICE_STATE ${wmStatePath}
+    _addWMCoreVenvVar WMCORE_SERVICE_LOGS ${wmLogsPath}
+    _addWMCoreVenvVar WMCORE_SERVICE_TMP ${wmTmpPath}
+    _addWMCoreVenvVar WMCORE_SERVICE_ROOT ${wmTopPath}
+
+    # add $wmSrcPath in front of everything if we are running from source
+    if $runFromSource; then
+        _addWMCoreVenvVar PYTHONPATH ${wmSrcPath}/src/python/:$PYTHONPATH
+        _addWMCoreVenvVar PATH ${wmSrcPath}/bin/:$PATH
+    fi
+}
+
+setupInitScripts(){
+    # Function to build the WMcore init scripts, based on the current setup
+    # configuration and set of enabled services
+    # Two types of scripts are created:
+    #  * `manage' script per enabled service, supporting basic operation like:
+    #     start, stop, restart, status
+    #  * `wmcmanage' global script for iterating through all enabled services. Supports:
+    #     start[:service], stop[:service], restart[:service], status[:service], version[:service]
+    # It uses the script's runtime prameters from global scope.
+    #:param: None
+    echo
+    echo "======================================================="
+    echo "Setup WMCore init scripts inside the virtual env:"
+    echo -n "Continue? [y]: "
+    $assumeYes || read x && [[ $x =~ (n|no|nO|N|No|NO) ]] && return 101
+    echo "..."
+
+    # DONE: To create the init.sh scripts
+    # First creating all the service level `manage' scripts:
+    local wmVersion=$(python -c "from WMCore import __version__ as WMCoreVersion; print(WMCoreVersion)")
+    for service in $enabledList
+    do
+        local manageScript=${wmCfgPath}/${service}/manage
+        [[ -d ${wmCfgPath}/${service} ]] && touch $manageScript && chmod 755 $manageScript || { err=$?; echo "could not setup startup scripts for $service";  return $err  ;}
+        cat<<EOF>$manageScript
+#!/bin/bash
```

Comment:
Can we avoid writing configuration file(s) from setup script? Can you commit this configuration file into some repo and use it? I think this approach is dangerous and will require lots of maintenance in a future. Instead all configs should be available through github/gitlab such that we can change them independently from this setup script.

Reply:
First, let me clarify something we are not writing config files here, but we are writing the `manage` scripts for the current services  deployed. This is exactly how it has been done from cmsweb deployment scripts for the `RPM` based deployment as  well. It is basically a pattern for a `manage` script provided here, and then it needs to be filled with a variable or two regarding the service at hand and then distributed at the right service area so that the global `wmcmanage` one can find every enabled service's `manage` script. I am talking about line [748-749](https://github.com/dmwm/WMCore/blob/9606ac57bd38ba3c6b758c4071b9f8a72ab27f8e/deploy/deploy-centralvenv.sh#L748-L749) and line [769](https://github.com/dmwm/WMCore/blob/9606ac57bd38ba3c6b758c4071b9f8a72ab27f8e/deploy/deploy-centralvenv.sh#L769) in the current version. And also the destination tied to the current deployment path on line [721](https://github.com/dmwm/WMCore/blob/9606ac57bd38ba3c6b758c4071b9f8a72ab27f8e/deploy/deploy-centralvenv.sh#L721). Those are variables defined at deployment time. If we can think of a proper way of providing them to the script (without using `sed`, which as already maentioned, I really find it far from elegant) those 'templates' can easily go to `WMCore/bin/.`  

Well for the `wmcmanage` one, we can already put it at `WMCore/bin`  as it is, because it does not depend on anything else but the list of `enabled` services. I have kept it here in order to have the whole process of creating all the `*manage` scripts at a single place.